src/twisted/words/test/test_irc.py

```diff
@@ -9,13 +9,27 @@
 import operator
 import time
 
+from twisted.internet import protocol, task
+from twisted.python.filepath import FilePath
+from twisted.python.compat import unicode
+from twisted.test.proto_helpers import StringTransport, StringIOWithoutClosing
 from twisted.trial import unittest
 from twisted.trial.unittest import TestCase
 from twisted.words.protocols import irc
 from twisted.words.protocols.irc import IRCClient, attributes as A
-from twisted.internet import protocol, task
-from twisted.test.proto_helpers import StringTransport, StringIOWithoutClosing
-from twisted.python.filepath import FilePath
+
+
+
+def assertEqualBufferValue(bufferValue, val):
+    # A buffer is always bytes, but sometimes
+    # we need to compare it to a utf-8 unicode string
+    if isinstance(val, unicode):
+        bufferValue = bufferValue.decode("utf-8")
+
+    if isinstance(bufferValue, list):
+        if isinstance(val[0], unicode):
+            bufferValue = [buf.decode("utf8") for buf in bufferValue]
+    assert bufferValue == val, "%s != %s" % (bufferValue, val)
```

Comment:
Assertions are generally frowned upon in Twisted, because [they aren't always run; `python -OO` disables them](https://docs.python.org/2/reference/simple_stmts.html#the-assert-statement).  This should be using `assertEqual`; which means this function should take a `TestCase` object as well.